File: 1524-string-matching-in-an-array/1524-string-matching-in-an-array.py
class Solution(object):
    def stringMatching(self, words):
        """
        :type words: List[str]
        :rtype: List[str]
        """
        res=[]
        n= len(words)
        for i, w in enumerate(words):
             for k in words[0:]:
                if w in k:
                    if w != k:
                      res.append(w)
                      break
           
        return res
        

File: 1524-string-matching-in-an-array/1524-string-matching-in-an-array.cpp
class Solution {
public:
    vector<string> stringMatching(vector<string>& words) {
        sort(words.begin(), words.end(), [](const string & a, const string &b){
            return a.size()<b.size();
        });
        for(auto x:words) cout<<x<<" ";
        vector<string>v;
        for(int i=0;i<words.size()-1;i++){
            string a=words[i];
            for(int j=i+1;j<words.size();j++){
                string b=words[j];
                // int k=0,l=0;
                if(b.find(a)!=string::npos){
                    v.push_back(a);
                    break;
                }
               
            }
        }
        return v;
    }
};